tensorboard/webapp/widgets/line_chart_v2/sub_view/line_chart_axis_view_test.ts

```diff
@@ -0,0 +1,164 @@
+import {By} from '@angular/platform-browser';
+
+import {createScale} from '../lib/scale';
+import {Extent, ScaleType} from '../lib/public_types';
+import {
+  LineChartXAxisComponent,
+  LineChartYAxisComponent,
+} from './line_chart_axis_view';
+
+@Component({
+  selector: 'testable-comp',
+  template: `
+    <line-chart-x-axis
+      [viewExtent]="viewBox"
+      [xScale]="scale"
+      [xGridCount]="10"
+      [domDim]="domDim"
+    ></line-chart-x-axis>
+    <line-chart-y-axis
+      [viewExtent]="viewBox"
+      [yScale]="scale"
+      [yGridCount]="5"
+      [domDim]="domDim"
+    ></line-chart-y-axis>
+  `,
+})
+class TestableComponent {
+  scale = createScale(ScaleType.LINEAR);
+
+  @Input()
+  viewBox: Extent = {
+    x: [100, 300],
+    y: [-1, 1],
+  };
+
+  domDim = {
+    width: 100,
+    height: 200,
+  };
+}
+
+describe('line_chart_v2/sub_view/axis test', () => {
+  const ByCss = {
+    X_AXIS_LABEL: By.css('line-chart-x-axis text'),
+    Y_AXIS_LABEL: By.css('line-chart-y-axis text'),
+  };
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [
+        TestableComponent,
+        LineChartXAxisComponent,
+        LineChartYAxisComponent,
+      ],
+      imports: [CommonModule],
+    }).compileComponents();
+  });
+
+  function assertLabels(debugElements: DebugElement[], axisLabels: string[]) {
+    const actualLabels = debugElements.map((el) =>
+      el.nativeElement.textContent.trim()
+    );
+    expect(actualLabels).toEqual(axisLabels);
+  }
+
+  function assertLabelLoc(
+    debugElements: DebugElement[],
+    expectedLocs: Array<{x: number; y: number}>
+  ) {
+    expect(debugElements.length).toBe(expectedLocs.length);
+    for (const [index, el] of debugElements.entries()) {
+      const actual = {x: el.attributes['x'], y: el.attributes['y']};
+      expect(actual).toEqual({
+        x: String(expectedLocs[index].x),
+        y: String(expectedLocs[index].y),
+      });
+    }
+  }
+
+  it('renders tick in human readable format', () => {
+    const fixture = TestBed.createComponent(TestableComponent);
+    fixture.detectChanges();
+
+    assertLabels(fixture.debugElement.queryAll(ByCss.X_AXIS_LABEL), [
+      '100',
+      '200',
+      '300',
+    ]);
+
+    assertLabels(fixture.debugElement.queryAll(ByCss.Y_AXIS_LABEL), [
+      '-1',
+      '-0.5',
+      '0',
+      '0.5',
+      '1',
+    ]);
+  });
+
+  it('updates to viewBox changes', () => {
+    const fixture = TestBed.createComponent(TestableComponent);
+    fixture.detectChanges();
+
+    fixture.componentInstance.viewBox = {x: [1e6, 5e6], y: [0, 1]};
+    fixture.detectChanges();
+
+    assertLabels(fixture.debugElement.queryAll(ByCss.X_AXIS_LABEL), [
+      '2e+6',
+      '4e+6',
+    ]);
+
+    assertLabels(fixture.debugElement.queryAll(ByCss.Y_AXIS_LABEL), [
+      '0',
+      '0.2',
+      '0.4',
+      '0.6',
+      '0.8',
+      '1',
+    ]);
+  });
+
+  it('aligns y axis to the right edge of its dom', () => {
```

Comment:
Screenshot should clarify the logic. In case it is not all that clear, these Angular components are laid out using CSS grid and take only partial region of the whole line chart.